src/Menu/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { borderRadius, color, space } from '@core-ds/primitives';
+import * as React from 'react';
+import styled from 'styled-components';
+import {
+   MenuButtonProps,
+   MenuContext,
+   MenuItemProps,
+   MenuListProps,
+   MenuProps,
+   useMenu,
+   useMenuButton,
+   useMenuItem,
+   useMenuList,
+} from './useMenu';
+
+export function Menu(props: MenuProps) {
+   const { children, className } = props;
+   const menu = useMenu(props);
+
+   return (
+      <MenuContext.Provider value={menu}>
+         <MenuContainer className={className}>{children}</MenuContainer>
+      </MenuContext.Provider>
+   );
+}
+
+const MenuContainer = styled.div`
+   position: relative;
+`;
+
+function UnstyledMenuButton(props: MenuButtonProps) {
+   const menuButtonProps = useMenuButton(props);
+   return <button {...menuButtonProps} />;
+}
+
+export const MenuButton = styled(UnstyledMenuButton)`
+   display: flex;
+   align-items: center;
+   background: none;
+   border: none;
+   color: ${color.white};
+   font-size: 14px;
+   cursor: pointer;
+   outline: none;
+   border-radius: ${borderRadius.md};
+   font-family: inherit;
+   padding: 0;
+   &:focus {
+      box-shadow: 0 0 0 0.2rem rgb(25 117 241 / 25%);
+   }
+`;
+
+export function UnstyledMenuList(props: MenuListProps) {
+   const menuListProps = useMenuList(props);
+
+   return <ul {...menuListProps} />;
+}
+
+export const MenuList = styled(UnstyledMenuList)`
+   margin: 0;
+   margin-bottom: ${space[1]};
+   padding: ${space[2]} 0;
+   border-radius: ${borderRadius.lg};
+   list-style: none;
+   background-color: white;
+   color: black;
+   display: none;
+   position: absolute;
+   transform: translateY(-100%) translateY(-${space[2]});
+   top: 0;
+   left: 0;
+   min-width: 224px;
```

Comment:
It's just a reasonable min width so that the menu looks good out of the box. The rationale behind this is that if you have a list with short items name, the menu will look ugly and you'll have extended anyway to define a min-width, e.g.

```jsx
const StyledMenu = styled(Menu)`
   min-width: ...
`
```

So, this is here just to provide a good default. Can be changed to whatever figure your design team thinks should be the minimum